### PR TITLE
Fixed a problem of adding padding of 4 zero bytes in some cases

### PR DESCRIPTION
**Pull request by nipuntalukdar**
Fixed a small problem of adding a padding of 4 zero bytes when the index file-size is already multiple of 4.

### CODE REVIEW

File: index.go [hunk 1]
```diff
@@ -203,10 +207,10 @@ func (w *indexWriter) write(bufs ...[]byte) error {
...
 }
 
-// addPadding adds zero byte padding until the file size is a multiple of n.
-func (w *indexWriter) addPadding(n int) error {
-	p := n - (int(w.pos) % n)
-	if p == 0 {
+// addPadding adds zero byte padding until the file size is a multiple size_unit.
+func (w *indexWriter) addPadding() error {
+	p := size_unit - (int(w.pos) & padding_and)
+	if p == size_unit {
```
Comment (fabxc) [inline index.go]: Would this be potentially simpler?

```go
p := w.pos % size_unit
if p == 0 {
    return nil
}
p = size_unit - p
```

That would skip the extra `padding_and` constant – `&` always need a few looks when reading it without context.

**Comment by fabxc** [comment]
Thanks, good catch! Added a comment that could potentially make it a bit simpler.

File: index.go (hunk 1 above)
Comment (nipuntalukdar) [reply index.go]: Okay. Will do the change. % is sometimes expensive, but anyway 4 is a very small number.

Comment (fabxc) [reply index.go]: It will probably be optimized into `&` by the compiler – but we are not calling too often in the grand scheme of things anyway :)

**Comment by fabxc** [comment]
Thanks 👍

**Pull request merged**